cscs-checks/tools/profiling_and_debugging/intel_sde_berkeley_stream.py

```diff
@@ -0,0 +1,153 @@
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class SdeBaseTest(rfm.RegressionTest):
+    '''This test checks the values reported by Intel SDE for roofline modeling:
+       - https://software.intel.com/en-us/articles/
+            intel-software-development-emulator
+       - https://bitbucket.org/dwdoerf/stream-ai-example/src/master/
+       - https://www.nersc.gov/
+            users/application-performance/measuring-arithmetic-intensity
+    '''
+    def __init__(self):
+        super().__init__()
+        self.descr = 'Roofline Analysis test with Intel SDE'
+        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
+                                       'roofline', 'sde')
+        self.build_system = 'SingleSource'
+        self.sourcepath = 'stream_mpi.c'
+        self.executable = 'sde'
+        self.target_executable = './stream.exe'
+        self.sde = '%s.sde' % self.target_executable
+        self.rpt = '%s.rpt' % self.target_executable
+        self.build_system.ldflags = ['-g', '-O3', '-qopenmp', '-restrict',
+                                     '-qopt-streaming-stores', 'always']
+        self.pre_run = [
+            'mv %s %s' % (self.executable, self.target_executable),
+            'module use /apps/dom/UES/jenkins/7.0.UP00/mc/easybuild/'
+            'experimental/modules/all',
```

Comment:
I'd say to put the whole path on the same line even if it exceeds the 80-column limit. It's more readable.